src/IECoreArnold/Renderer.cpp

```diff
@@ -3083,14 +3083,26 @@ class ArnoldGlobals
 			if( m_messageHandler )
 			{
 				m_messageCallbackId = AiMsgRegisterCallback( &messageCallback, m_consoleFlags, this );
+#if ARNOLD_VERSION_NUM < 70100
 				AiMsgSetConsoleFlags( m_renderSession.get(), AI_LOG_NONE );
+#else
+				AiMsgSetConsoleFlags( m_universeBlock->universe(), AI_LOG_NONE );
+#endif
 			}
 			else
 			{
+#if ARNOLD_VERSION_NUM < 70100
 				AiMsgSetConsoleFlags( m_renderSession.get(), m_consoleFlags );
+#else
+				AiMsgSetConsoleFlags( m_universeBlock->universe(), m_consoleFlags );
+#endif
 			}
 
+#if ARNOLD_VERSION_NUM < 70100
 			AiMsgSetLogFileFlags( m_renderSession.get(), m_logFileFlags );
+#else
+			AiMsgSetLogFileFlags( m_universeBlock->universe(), m_logFileFlags );
```

Comment:
Based on the documentation, I think we also need to update the [message callback](https://github.com/GafferHQ/gaffer/blob/main/src/IECoreArnold/Renderer.cpp#L3860-L3871) to check for a `universe` metadata rather than a `render_session` one. Currently I think `IECoreArnoldTest.RendererTest.testMessageHandler` is passing only because we're not filtering messages at all, rather than because we're filtering them properly.

Reply:
I _think_ I implemented this as expected, but the test is still failing on the `SceneDescription` mode where the ass writing messages seem to still be directed to another (presumably default) universe. It did reduce the message count on the batch & interactive modes by 2 messages, but I don't really understand why those 2 are directed back to the default universe:

```
00:00:00   385MB         |  
00:00:00   385MB         | releasing resources
```

Reply:
I had hoped that this "messages go to the default universe" bug would have been fixed in 7.1 - I thought that was one of the motivations for the breaking changes in the API. Since those look like the last two messages, is it possible that we've removed our message handler too early?

Reply:
Oh wait, maybe the last 2 batch/interactive messages being filtered does make sense... 

https://github.com/GafferHQ/gaffer/blob/main/src/IECoreArnold/Renderer.cpp#L3112-L3120

We're resetting the render session & universe while the handler is active, and then the check for "does the universe in the metadata match _that_ universe" is going to say no (because that universe is a nullptr now) and the last shutdown messages get filtered out.

This still doesn't explain why in `SceneDescription` mode the following messages also go to the default universe _before_ those 2 pointers were reset:

```
00:00:01   389MB         |  [ass] writing scene to /tmp/gafferTestspkjwM/test.ass (mask=0xFFFF) ...
00:00:01   389MB         |  [ass] wrote 892 bytes, 2 nodes in 0:00.00
```

Reply:
Note: I edited the comment above significantly _after_ the email posted...

Reply:
I've confirmed with Arnold that this is still broken in 7.1, so I've done the following:

- (98459aa1f55698c2a804fb496747c91874a5ddc3) Added the message callback handling (sorry I squashed this in before realizing I hadn't pushed it online yet)
- (bdbeafe34254086f8c264860aa441092ec29cee7) Copied the changes from #4556
- (9ee187dc4a2d0c1ef47a7acdf4280cab60d6f5dd) Limited the `skipTest` to the SceneDescription mode in Arnold 7

With this the Arnold tests are passing in 6.2.1.1, 7.0.0.2, and 7.1.0.0RC2 at IE.